packages/core/src/jsx-runtime.js

```diff
@@ -0,0 +1,174 @@
+
+  const newProps = {}
+  for (let key in props) {
+    if (
+      hasOwnProperty.call(props, key) &&
+      key !== 'css' &&
+      key !== typePropName &&
+      (process.env.NODE_ENV === 'production' || key !== labelPropName)
+    ) {
+      newProps[key] = props[key]
+    }
+  }
+  newProps.ref = ref
+  newProps.className = className
+
+  const ele = jsx(type, newProps)
+  if (!isBrowser && rules !== undefined) {
+    let serializedNames = serialized.name
+    let next = serialized.next
+    while (next !== undefined) {
+      serializedNames += ' ' + next.name
+      next = next.next
+    }
+    return (
+      <React.Fragment>
+        <style
+          {...{
+            [`data-emotion-${cache.key}`]: serializedNames,
+            dangerouslySetInnerHTML: { __html: rules },
+            nonce: cache.sheet.nonce
+          }}
+        />
+        {ele}
+      </React.Fragment>
+    )
+  }
+  return ele
+}
+
+let Emotion = /* #__PURE__ */ withEmotionCache((props, cache, ref) => {
+  // use Context.read for the theme when it's stable
+  if (typeof props.css === 'function') {
+    return (
+      <ThemeContext.Consumer>
+        {theme => render(cache, props, theme, ref)}
+      </ThemeContext.Consumer>
+    )
+  }
+  return render(cache, props, null, ref)
+})
+
+if (process.env.NODE_ENV !== 'production') {
+  Emotion.displayName = 'EmotionCssPropInternal'
+}
+
+const jsx = (type, props, maybeKey) => {
+  let args = arguments
+
+  if (props == null || !hasOwnProperty.call(props, 'css')) {
+    return reactJsx(type, props, maybeKey)
+  }
+
+  if (
+    process.env.NODE_ENV !== 'production' &&
+    typeof props.css === 'string' &&
+    // check if there is a css declaration
+    props.css.indexOf(':') !== -1
+  ) {
+    throw new Error(
+      `Strings are not allowed as css prop values, please wrap it in a css template literal from '@emotion/css' like this: css\`${
+        props.css
+      }\``
+    )
+  }
+
+  let argsLength = args.length
+
+  let createElementArgArray = new Array(argsLength)
+
+  createElementArgArray[0] = Emotion
+  let newProps = {}
+
+  for (let key in props) {
+    if (hasOwnProperty.call(props, key)) {
+      newProps[key] = props[key]
+    }
+  }
+  newProps[typePropName] = type
+  if (process.env.NODE_ENV !== 'production') {
+    let error = new Error()
+    if (error.stack) {
+      // chrome
+      let match = error.stack.match(
+        /at (?:Object\.|Module\.|)jsx.*\n\s+at ([A-Z][A-Za-z$]+) /
+      )
+      if (!match) {
+        // safari and firefox
+        match = error.stack.match(/.*\n([A-Z][A-Za-z$]+)@/)
+      }
+      if (match) {
+        newProps[labelPropName] = sanitizeIdentifier(match[1])
+      }
+    }
+  }
+
+  createElementArgArray[1] = newProps
+
+  for (let i = 2; i < argsLength; i++) {
+    createElementArgArray[i] = args[i]
+  }
+  // $FlowFixMe
+  return reactJsx.apply(null, createElementArgArray)
+}
+
+const jsxs = jsx
+
+export { jsx, jsxs, Fragment, createElement }
```

Comment:
`createElement` not need to export in `@emotion/core/jsx-runtime`

if have to, need to export in `@emotion/core`. but i think this may some order issue of babel.

result by my custom `jsx-runtime`

![image](https://user-images.githubusercontent.com/1667873/96073347-d8536180-0ed8-11eb-918b-fa670b73ebac.png)

Reply:
got it, codes like `<span {...v} key={i}>{i}</span>` will be use `createElement` of `@jsxImportSource`

@babel/plugin-transform-react-jsx: ^7.10.4

```jsx
const SomeComponent = () => {
  return (
    <>
      <div>123</div>
      <span>{undefined}</span>
      <>
        {[{}, {}].map((v, i) => <span {...v} key={i}>{i}</span>)}
      </>
    </>
  );
};

// ↓ ↓ ↓ ↓ ↓ ↓

const SomeComponent = () => {
  return /*#__PURE__*/_jsxs(_Fragment, {
    children: [/*#__PURE__*/_jsx("div", {
      children: "123",
    }), /*#__PURE__*/_jsx("span", {
      children: undefined,
    }), /*#__PURE__*/_jsx(_Fragment, {
      children: [{}, {}].map((v, i) => /*#__PURE__*/_createElement("span", {
        ...v,
        key: i,
      }, i)),
    })],
  });
}; 
```

Reply:
https://github.com/babel/babel/blob/master/packages/babel-helper-builder-react-jsx-experimental/src/index.js#L270
```
  // We want to use React.createElement, even in the case of
  // jsx, for <div {...props} key={key} /> to distinguish it
  // from <div key={key} {...props} />. This is an intermediary
  // step while we deprecate key spread from props. Afterwards,
  // we will stop using createElement in the transform.
```

may no need to export the `createElement`